TombEngine/Game/camera.cpp

```diff
@@ -405,6 +405,16 @@ void MoveObjCamera(GameVector* ideal, ItemInfo* camSlotId, int camMeshId, ItemIn
 	}
 }
 
+void RefreshFixedCamera(short camNumber)
+{
+	auto* camera = &g_Level.Cameras[camNumber];
+
+	GameVector from = GameVector(camera->Position, camera->RoomNumber);
+	int moveSpeed = camera->Speed * 8 + 1;
```

Comment:
Demagic numbers?

Reply:
I'm not sure how.

The other line where this happens (line 874) had this comment:

> // Multiply original speed by 8 to comply with original bitshifted speed from TR1-2

But I'm not very sure how the speed is stored in the camera (and is even possible stoe speed in the camera with TEN? in Tomb Editor that field appears disabled in grey to me).

I let it like that to ensure I don't break anything. :(

Reply:
For now you could define a constant at the top of the function:
`constexpr auto SPEED_COEFF = 8;`

Reply:
The 8 itself is not a magic number nor a variable that can be placed as anyone likes. Is mechanical. What it's doing is bitshifting the camera speed 3 bits to the left.  

(So if TE camera.speed is 00000110 = 6  * 8 = 48 = 00110000)

What I don't understand is why TE camera has a value that needs to be displaced like this to be used.